### PR TITLE
JP-2579: Add NIRSpec lamp processing info to spec2 pipeline description

### PR DESCRIPTION
**Pull request by stscirij**
Resolves [JP-2579](https://jira.stsci.edu/browse/JP-2579)

**Description**

Add information on processing NIRSpec lamp exposures to the Spec2pipeline documentation

Checklist
- [ ] Tests
- [x] Documentation
- [x] Change log
- [x] Milestone
- [x] Label(s)

### CODE REVIEW

File: docs/jwst/pipeline/calwebb_spec2.rst [hunk 1]
```diff
@@ -96,6 +96,110 @@ performed on the original (unresampled) data. The :ref:`cube_build <cube_build_s
...
+| :ref:`photom <photom_step>`           |  NONE      |    NONE      |      NONE       |      NONE    |
++---------------------------------------+------------+--------------+-----------------+--------------+
+| :ref:`resample_spec <resample_step>`  |  MOS, FS   |    NONE      |    MOS, FS      |      NONE    |
++---------------------------------------+------------+--------------+-----------------+--------------+
+| :ref:`cube_build <cube_build_step>`   |   IFU      |    NONE      |       IFU       |      NONE    |
++---------------------------------------+------------+--------------+-----------------+--------------+
+| :ref:`extract_1d <extract_1d_step>`   |   ALL      |    NONE      |       ALL       |      NONE    |
++---------------------------------------+------------+--------------+-----------------+--------------+
+
+In the resample_spec and cube build steps, the spectra are transformed to a space of (wavelength,
```
Comment (hbushouse) [inline docs/jwst/pipeline/calwebb_spec2.rst:200]: Could put back-tics around the resample_spec and cube_build step names. Or perhaps even make them links (references) to the steps.

File: docs/jwst/pipeline/calwebb_spec2.rst [hunk 2]
```diff
@@ -96,6 +96,110 @@ performed on the original (unresampled) data. The :ref:`cube_build <cube_build_s
...
 
+NIRSpec Lamp Exposures
+----------------------
+The ``Spec2Pipeline`` works slightly differently for NIRSpec lamp exposures.
+These are identified by the EXP_TYPE values of NRS_LAMP, NRS_AUTOWAVE or
+NRS_AUTOFLAT.  Using the EXP_TYPE keyword in this way means that another keyword
+is needed to specify whether the data are Fixed Slit, MOS, IFU or Brightobj.
+This is the OPMODE keyword, which maps to the ``jwst.datamodel`` attribute
+``.meta.instrument.lamp_mode``.  This keyword can take the following values in
+in exposures that undergo ``Spec2Pipeline`` processing:
```
[the remark this answers is not in the document]
Comment (stscirij) [reply docs/jwst/pipeline/calwebb_spec2.rst:107]: Good catch!

[the remark this answers is not in the document]
Comment (stscirij) [reply docs/jwst/pipeline/calwebb_spec2.rst]: Fixed

File: docs/jwst/pipeline/calwebb_spec2.rst (hunk 1 above)
Comment (stscirij) [reply docs/jwst/pipeline/calwebb_spec2.rst]: Good suggestion, fixed

File: docs/jwst/pipeline/calwebb_spec2.rst [hunk 3]
```diff
@@ -96,6 +96,110 @@ performed on the original (unresampled) data. The :ref:`cube_build <cube_build_s
 a resampled/rectified cube for IFU exposures, which is then used as input to
 the :ref:`extract_1d <extract_1d_step>` step.
 
+NIRSpec Lamp Exposures
```
[the remark this answers is not in the document]
Comment (stscirij) [reply docs/jwst/pipeline/calwebb_spec2.rst:102]: Another good suggestion, implemented in latest commit

**Review by stscieisenhamer — APPROVED** [review]

**Review by hbushouse — APPROVED** [review]
Looks good.

**Pull request merged**